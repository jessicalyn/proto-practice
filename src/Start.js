import React, {Component} from 'react';

export default class Start extends Component {

  render = () => {
    return (
      <div>
        <h2>How to Play</h2>
        <h3>For each question you will see an image of a prototype problem with the prototype name missing. Choose the correct prototype from the answers listed below.</h3>
        <h3>Answers missed with be added to your study guide for futher practice.</h3>
        <button className="start-practice large-button" onClick={this.props.startPractice}>Start Practicing</button>
      </div>
    )
  }
}