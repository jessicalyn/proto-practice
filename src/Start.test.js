import React from 'react';
import Start from './Start.js';
import { shallow } from 'enzyme';

const startPracticeMock = jest.fn();

describe('End', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Start
        startPractice={startPracticeMock}
      />
    )
  });

  it('should match a snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should fire start practice method when clicked', () => {
    wrapper.find('.start-practice').simulate('click');
    expect(startPracticeMock).toBeCalled();
  });

})