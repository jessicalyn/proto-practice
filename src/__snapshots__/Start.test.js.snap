// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End should match a snapshot with all data passed in 1`] = `
<div>
  <h2>
    How to Play
  </h2>
  <h3>
    For each question you will see an image of a prototype problem with the prototype name missing. Choose the correct prototype from the answers listed below.
  </h3>
  <h3>
    Answers missed with be added to your study guide for futher practice.
  </h3>
  <button
    className="start-practice large-button"
    onClick={[MockFunction]}
  >
    Start Practicing
  </button>
</div>
`;
